open-widget-framework/src/defaults.js

```diff
@@ -20,34 +20,69 @@ class _defaultRenderer extends Component {
 }
 
 class _defaultListWrapper extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      editMode: false,
+      addWidgetForm: false,
+      editWidgetForm: null,
+    }
+    this.renderAddWidgetButton = this.renderAddWidgetButton.bind(this)
+    this.closeForm = this.closeForm.bind(this)
+  }
+
   renderAddWidgetButton() {
     return (
-      <button className={'btn btn-info'} onClick={this.props.openNewWidgetForm}>
+      <button className={'btn btn-info'} onClick={() => this.setState({
+        addWidgetForm: !this.state.addWidgetForm,
+        editWidgetForm: null,
+      })}>
         <Octicon name={'plus'}/>
       </button>
     )
   }
+
+  closeForm() {
+    this.setState({
+      addWidgetForm: false,
+      editWidgetForm: null,
+    })
+  }
+
   render() {
     return (
-      <div>
-        {this.props.renderWidgetForm()}
+      <div className={'bg-secondary rounded float-right col-lg-4'}>
+        {this.state.addWidgetForm ? this.props.renderNewWidgetForm({closeForm: this.closeForm}) : null}
+        {this.state.editWidgetForm !== null
+          ? this.props.renderEditWidgetForm(this.state.editWidgetForm, {closeForm: this.closeForm}) : null}
         <div className={'edit-widget-list-bar btn-group'} role={'group'}>
-          <button className={'btn btn-info' + (this.props.editModeActive ? ' active' : '') }
-                  onClick={this.props.toggleEditMode}>
+          <button className={'btn btn-info' + (this.state.editMode ? ' active' : '') }
```

Comment:
```suggestion
          <button className={`btn btn-info ${this.state.editMode ? "active" : "")`}
```
Assuming nwb babel config supports string interpolation.